File: java/src/pattern/strategy/adventure/Weapon.java
package pattern.strategy.adventure;

public abstract class Weapon
{
    protected String name;
    protected WeaponBehaviour behaviour;

    public void setName(String name)
    {
	this.name = name;
    }

    public void setBehaviour(WeaponBehaviour behaviour)
    {
	this.behaviour = behaviour;
    }

    public String fight()
    {
	return behaviour.toString();
    }

    public String toString()
    {
	return name;
    }
}

File: php/src/pattern/strategy/adventure/Weapon.php
<?php

namespace pattern\strategy\adventure;

abstract class Weapon
{
    protected $_name;
    protected $_behaviour;

    public function __construct()
    {
    }

    public function setName($name)
    {
	$this->_name = $name;
    }

    public function setBehaviour(WeaponBehaviour $behaviour)
    {
	$this->_behaviour = $behaviour;
    }

    public function fight()
    {
	return (string) $this->_behaviour;
    }

    public function __toString()
    {
	return $this->_name;
    }

}
